### DISCUSSION TITLE
Should we color global variables in lua differently than regular variables

### DISCUSSION THREAD

**Discussion by tristan957** [olimorris/onedarkpro.nvim#246] [Color match]
Here is the output of `:Inspect` when hovering the `vim` global variable.

```text
Treesitter
  - @variable.lua links to @variable lua

Semantic Tokens
  - @lsp.type.variable.lua links to @variable priority: 125
  - @lsp.mod.global.lua links to @lsp priority: 126
  - @lsp.typemod.variable.global.lua links to @lsp priority: 127

Extmarks
  - LspReferenceRead links to LspReferenceText vim_lsp_references
  - LspReferenceRead links to LspReferenceText vim_lsp_references
```

This would seemingly be a departure from the VSCode theme, best I can tell, but I think it would help with readability. This would be inline with how we highlight `string`, `table`, `io`, etc.

**Comment by olimorris** [comment]
Can you share some screenshots?

**Reply by tristan957** [reply]
Old:
![image](https://github.com/user-attachments/assets/5425d22d-39e0-47a0-b882-c7899210bfa8)

New:
![image](https://github.com/user-attachments/assets/73314e87-ad74-4de0-93ef-ac078a25d48c)

Settings:

```lua
["@lsp.typemod.variable.global.lua"] = { link = "@lsp.typemod.variable.defaultLibrary.lua" },
```

**Reply by tristan957** [reply]
I don't know if this should only apply to lua or all `@lsp.typemod.variable.global`

**Reply by olimorris** [reply]
Have you got any settings applied in VS Code? Mine looks like:

![2024-08-01 17_03_32 - Code@2x](https://github.com/user-attachments/assets/8fd6ec17-740c-438f-85fc-777486f5885a)

**Reply by tristan957** [reply]
That is also how it looks in vscode for me.

> This would seemingly be a departure from the VSCode theme

**Reply by olimorris** [reply]
Sorry I missed your comment earlier. As this deviates from the VS Code theme I'll keep it out.

Playing around though, setting a custom highlight of `["@lsp.typemod.variable.global.lua"] = { fg = "${yellow}" }` gives:

![2024-08-01 17_35_31 - WezTerm@2x](https://github.com/user-attachments/assets/e659a9ce-b547-4f87-aa98-d7856345d8e5)

Otherwise, the only option would be some custom tree-sitter queries to pick out variables like "vim" etc.

**Reply by olimorris** [reply]
@tristan957 if you have a list of keywords that you think should be highlighted, I'll be happy to put a tree-sitter query in the plugin that detects these because I do agree that it helps readability and breaks up the chain of red

**Reply by tristan957** [reply]
No, it's all good. I'll just keep this highlight locally then.
